sql/debug_sync.cc

```diff
@@ -1975,6 +1975,11 @@ bool debug_sync_set_action(THD *thd, const char *action_str, size_t len) {
   assert(thd);
   assert(action_str);
 
+  // A zero value keeps the facility disabled.
```

Comment:
I guess MTR framework always has debug sync enabled? Is it possible to use different cnf or start mysqld without debug sync enabled and execute testcase provided in the bug report?